core/capabilities/remote/target/receiver.go

```diff
@@ -0,0 +1,129 @@
+package target
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+	"time"
+
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/target/request"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type receiverRequest interface {
+	OnMessage(ctx context.Context, msg *types.MessageBody) error
+	Expired() bool
+	Cancel(err types.Error, msg string) error
```

Comment:
why is this asymmetric WRT the callerRequest?

Reply:
removed